src/cmds/scripts/pbs_habitat.in

```diff
@@ -242,9 +242,9 @@ upgrade_pbs_database() {
 	sys_pgsql_ver=$(echo `${PGSQL_DIR}/bin/psql -V` | awk '{print $NF}' | cut -d '.' -f 1,2)
 	ver=`cat ${data_dir}/PG_VERSION`
 
-	result=1
-	[[ $ver -ge $sys_pgsql_ver ]] && result=0
-	if [ ${result} -eq 1 ]; then
+	[ ${ver%.*} -eq ${sys_pgsql_ver%.*} ] && [ ${ver#*.} \> ${sys_pgsql_ver#*.} ] || [ ${ver%.*} -gt ${sys_pgsql_ver%.*} ];
+	result=$?
+	if [ ${result} -eq 0 ]; then
```

Comment:
@nithinj Added a new commit addressing the fact that upgrade_db routine should be called only if version is different.